File: from-3301-to-3400/3345-smallest-divisible-digit-product-i/solution.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        def getProduct(num: int) -> int:
            product = 1
            while num > 1:
                product *= (num % 10)
                num //= 10
            return product

        for c in range(t):
            if getProduct(n + c) % t == 0:
                return n + c

        return -1

File: from-3301-to-3400/3345-smallest-divisible-digit-product-i/solution.go
func smallestNumber(n int, t int) int {
    getProduct := func(n int) int {
        p := 1
        for n > 1 {
            p *= n % 10
            n /= 10
        }
        return p
    }
    for c := 0; c < t; c ++ {
        if getProduct(n + c) % t == 0 {
            return n + c
        }
    }
    return -1
}